src/dev/precommit_hook/check_dev_docs.js

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+const SIDEBAR_PATH = 'docs/_sidebar.md';
+
+export async function checkDevDocs(log, files) {
+  files.map((file) => {
+    const path = file.getRelativePath();
+
+    if (path === SIDEBAR_PATH) {
+      throw Error(`Please git add the ${SIDEBAR_PATH} before committing your changes to update the developer docs`);
```

Comment:
```suggestion
      throw Error(`The ${SIDEBAR_PATH} file of the developer docs has been modified but is not ready to be committed. This can be done by performing "git add ${SIDEBAR_PATH}" and committing the changes.`);
```